File: rb/53.rb
def fact n
  return 1 if n < 2
  (2..n).reduce(:*)
end

total = 0

(1..100).each do |n|
  (1..n).each do |r|
    denom = fact(r) * fact(n - r)
    total += 1 if fact(n) / denom > 1e6
  end
end

p total

File: py/53.py
def fact(n):
    return 1 if n < 2 else n * fact(n - 1)

total = 0

for n in xrange(1, 101):
    for r in xrange(1, n + 1):
        denom = fact(r) * fact(n - r)
        if fact(n) / denom > 1e6:
            total += 1

print total
